pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -83,6 +83,47 @@ func TestMirrors(t *testing.T) {
 	assert.True(t, reg.Mirrors[1].Insecure)
 }
 
+func TestSubdomainRefMatchesPrefix(t *testing.T) {
+	for _, c := range []struct {
+		ref, prefix string
+		expected    bool
+	}{
+		// Check for subdomain matches
+		{"docker.io", "*.io", true},
+		{"docker.io", "*.com", false},
+		{"example.com:5000", "*.com:5000", true},
+		{"example.com:50000", "*.example.com:5000", false},
+		{"example.com/foo", "*.com", true},
+		{"example.com/foo", "*.com/foo", false},
+		{"example.com/foo/bar", "*.com", true},
+		{"example.com/foo/bar", "*.com/foo", true},
+		{"example.com/foo/bar/bbq:baz", "*.com", true},
+		{"example.com/foo/bar/bbq:baz", "*.com/foo", true},
+		{"example.com/foo/bar/bbq:baz", "*.com/foo/bar", true},
+		{"example.com/foo/bar/bbq:baz", "*.com/foo/bar/bbq", false},
```

Comment:
Same here. Why is that not match?

Reply:
yup, that's old stuff there, will be changed.